src/Controls/src/Core/VisualElement.cs

```diff
@@ -23,9 +23,11 @@ public partial class VisualElement : NavigableElement, IAnimatable, IVisualEleme
 		/// <include file="../../docs/Microsoft.Maui.Controls/VisualElement.xml" path="//Member[@MemberName='InputTransparentProperty']/Docs/*" />
 		public static readonly BindableProperty InputTransparentProperty = BindableProperty.Create("InputTransparent", typeof(bool), typeof(VisualElement), default(bool));
 
+		internal static readonly BindableProperty UnCoercedIsEnabledProperty = BindableProperty.Create("UnCoercedIsEnabled", typeof(bool), typeof(VisualElement), true);
+
 		/// <include file="../../docs/Microsoft.Maui.Controls/VisualElement.xml" path="//Member[@MemberName='IsEnabledProperty']/Docs/*" />
 		public static readonly BindableProperty IsEnabledProperty = BindableProperty.Create("IsEnabled", typeof(bool),
-			typeof(VisualElement), true, propertyChanged: OnIsEnabledPropertyChanged);
+			typeof(VisualElement), true, propertyChanged: OnIsEnabledPropertyChanged, coerceValue: CoerceIsEnabledProperty);
```

Comment:
Maybe I am holding it wrong, but one limitation of Maui is that it does not preserve the previous value. I was browsing WPF source and they have a feature to re-evaluate a property using the original, pre-coerced value. Not sure we have one like that, but I just used a hidden property... Probably bad.